blockchain/emulator_test.go

```diff
@@ -59,5 +63,123 @@ func Test_TranslateAddress(t *testing.T) {
+
+	fmt.Println("Validating account storage...")
+	for i := 0; i < testAccounts; i++ {
+		_, accountStorage, err := emu.getAccount(accountList[i].Address)
+		assert.NoError(t, err)
+		assert.Equal(t, accountStorage.Account.Address.String(), accountList[i].Address.String())
+		assert.Equal(t, accountStorage.Account.Address.Hex(), accountList[i].Address.Hex())
+		assert.Equal(t, accountStorage.Account.Address.Bytes(), accountList[i].Address.Bytes())
+	}
+}
+
+// Test_DeployEmptyContract tests deployment of an empty contract
+func Test_DeployEmptyContract(t *testing.T) {
+	emu, err := newEmulator()
+	assert.NoError(t, err)
+	account, _, _, err := emu.createAccount()
+	_, _, err = emu.deployContract(account.Address, "")
+	assert.Error(t, err)
+}
+
+// Test_DeployBasicContracts tests deployment of a large number of basic contracts to a single account
+func Test_DeployBasicContracts(t *testing.T) {
+	emu, err := newEmulator()
+	assert.NoError(t, err)
+	account, _, _, err := emu.createAccount()
+
+	const numContracts int = 1000
+
+	const baseName string = "Foo"
+	var deployedContracts []string
+
+	for i := 0; i < numContracts; i++ {
+		name := baseName + strconv.Itoa(i)
+		contract := "pub contract " + name + "{}"
+		deployedContracts = append(deployedContracts, name)
+
+		_, tx, err := emu.deployContract(account.Address, contract)
+		assert.NoError(t, err)
+		assert.Equal(t, tx.Authorizers[0], account.Address)
+	}
+
+	account, _, err = emu.getAccount(account.Address)
+
+	keys := make([]string, 0, len(account.Contracts))
+	for k := range account.Contracts {
+		keys = append(keys, k)
+	}
+
+	// Verify that every deployed contract is included
+	for _, deployed := range deployedContracts {
+		contains := false
+		for _, contract := range keys {
+			if contract == deployed {
+				contains = true
+				break
+			}
+		}
+		assert.Equal(t, contains, true)
+	}
+}
+
+// Test_ParseContractName tests contract name parsing returns the correct name
+func Test_ParseContractName(t *testing.T) {
+	contract := "pub contract foo {}"
+	name, err := parseContractName(contract)
+	assert.NoError(t, err)
+	assert.Equal(t, name, "foo")
+
+	longName := "foo"
+	for i := 0; i < 100000; i++ {
+		longName += "long"
+	}
+
+	contract = "pub contract " + longName + " {}"
+	name, err = parseContractName(contract)
+	assert.NoError(t, err)
+	assert.Equal(t, name, longName)
+
+	contract = "pub contract foo_bar {}"
+	name, err = parseContractName(contract)
+	assert.NoError(t, err)
+	assert.Equal(t, name, "foo_bar")
+
+	// Double name
+	contract = "pub contract foo bar {}"
+	name, err = parseContractName(contract)
+	assert.Error(t, err)
+
+	// No name
+	contract = "pub contract {}"
+	name, err = parseContractName(contract)
+	assert.Error(t, err)
+
+	// Decimal name
+	contract = "pub contract 123foo {}"
+	name, err = parseContractName(contract)
+	assert.Error(t, err)
 
+	// Invalid character
+	contract = "pub contract foo! {}"
+	name, err = parseContractName(contract)
+	assert.Error(t, err)
```

Comment:
These are good candidates for table tests, [you can read about them here](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests), they are a good Go practice.